### PR TITLE
weak etag match support

### PR DESCRIPTION
**Pull request by lqqyt2423**
https://github.com/fastify/fastify-etag/issues/108

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ async function fastifyEtag (app, opts) {
       reply.header('etag', etag)
     }
 
-    if (req.headers['if-none-match'] === etag) {
+    if (req.headers['if-none-match'] === etag || req.headers['if-none-match'] === 'W/' + etag || 'W/' + req.headers['if-none-match'] === etag) {
```
Comment (Uzlopak) [inline index.js:51]: I have somehow the feeling that we can optimize this.

we have opts.weak so we know if our generated etag is weak or not.

Comment (lqqyt2423) [reply index.js:51]: Please see the issue: https://github.com/fastify/fastify-etag/issues/108

**Review by mcollina — APPROVED** [review]
lgtm

File: index.js (hunk 1 above)
Comment (Uzlopak) [reply index.js:51]: but you have etag. So you could check if it is a WeakTag. like 

```javascript
function isWeakTag(value) {
  return (
    value[0] === 'W' &&
    value[1] === '\'
  )
}
```

```suggestion
    const ifNoneMatch = req.headers['if-none-match']

    if (ifNoneMatch === etag || (isWeakTag(etag) ? 'W/' + ifNoneMatch === etag : ifNoneMatch === 'W/' + etag)) {
```

Comment (Uzlopak) [reply index.js:51]: I will benchmark and see if it has performance improvements

**Review by Uzlopak — CHANGES_REQUESTED** [review]
I want to make benchmarks and see if we can gain some perf.

**Comment by maidul98** [comment]
Having the same issue. Can you please get this merged in?

**Review by gurgunday — COMMENTED** [review]
@lqqyt2423, there are conflicts, can you take a look?

**Comment by lqqyt2423** [comment]
@gurgunday Resolved conflict

**Review by gurgunday — APPROVED** [review]
lgtm

**Pull request merged**